core/src/main/java/com/sequenceiq/cloudbreak/service/image/userdata/UserDataService.java

```diff
@@ -84,8 +89,9 @@ public void createUserData(Long stackId) throws CloudbreakImageNotFoundException
         try {
             PlatformParameters platformParameters = platformParametersFuture.get();
             CcmParameters ccmParameters = fetchCcmParameters(stack);
+            Optional<ProxyConfig> proxyConfig = proxyConfigDtoService.getByEnvironmentCrn(stack.getEnvironmentCrn());
             Map<InstanceGroupType, String> userData = userDataBuilder.buildUserData(Platform.platform(stack.getCloudPlatform()), cbSshKeyDer,
-                    sshUser, platformParameters, saltBootPassword, cbCert, ccmParameters);
+                    sshUser, platformParameters, saltBootPassword, cbCert, ccmParameters, proxyConfig.orElse(null));
```

Comment:
we should just pass the optional here and not null

Reply:
Finally we agreed with @lacikaaa to not use optional, as it is not recommended and Idea warns about it as well...